pennylane/ops/op_math/decompositions/single_qubit_unitary.py

```diff
@@ -185,7 +185,8 @@ def _rot_decomposition(U, wire, return_global_phase=False):
     """
 
     # Cast to batched format for more consistent code
-    U = math.expand_dims(U, axis=0) if len(U.shape) == 2 else U
+    if not sp.sparse.issparse(U):
+        U = math.expand_dims(U, axis=0) if len(U.shape) == 2 else U
```

Comment:
`expand_dims` will leverage the 2-axes mat to 3-axes tensors here, which we do not expect to be feasible for `scipy.sparse` (at least for now).